update URL mapping and add advprop pretrain models

Cyril9227 opened the pull request:
@toandaominh1997 thank's for your amazing work. 

However the URL links are broken, this PR fixes the URL mapping and also provides the option to load pretrained models using adversarial training, called advprop, see [luke melas implementation](https://github.com/lukemelas/EfficientNet-PyTorch).

Also modified docstrings and comments in efficientnet.py so they reflect better what the code is actually doing (it was previously a copy paste from another repo) and made few changes so the code stay consistent with the new available weights.

The project should be operational with those changes.

Cyril9227 committed:
update URL map and add advprop pretrain models

Cyril9227 committed:
Merge pull request #1 from Cyril9227/Cyril9227-update-url-map

update URL map and add advprop pretrain models

pep8speaks commented:
Hello @Cyril9227! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`models/efficientnet.py`](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/efficientnet.py):

> [Line 49:1](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/efficientnet.py#L49): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 180:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/efficientnet.py#L180): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 194:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/efficientnet.py#L194): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 216:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/efficientnet.py#L216): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (121 > 79 characters)
> [Line 228:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/efficientnet.py#L228): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 250:68](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/efficientnet.py#L250): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

* In the file [`models/utils.py`](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py):

> [Line 307:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L307): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 308:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L308): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 309:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L309): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 310:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L310): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 311:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L311): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 312:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L312): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 313:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L313): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 314:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L314): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 319:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L319): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 319:116](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L319): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 320:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L320): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 321:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L321): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 322:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L322): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 323:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L323): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 324:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L324): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 325:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L325): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 326:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L326): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 327:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L327): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 342:80](https://github.com/toandaominh1997/EfficientDet.Pytorch/blob/b945475dae09d27ef8835fa568fff6b7886a4809/models/utils.py#L342): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 79 characters)



##### Comment last updated at 2020-01-28 07:51:51 UTC

Cyril9227 committed:
Update efficientnet.py

Removed duplicate function from_pretrained and added advprop option to from_pretrained function

Cyril9227 committed:
Consistency update

Modified docstrings and comments so they reflect better what the code is actually doing (it was previously a copy paste from another repo) and made few changes so the code stay consistent with the new available weights

Cyril9227 committed:
Updated function signature

JVGD commented:
Please merge this, I am having the same problem. 
By the way, thank you @Cyril9227
